pkg/services/secrets/kvstore/helpers.go

```diff
@@ -29,6 +33,26 @@ func SetupTestService(t *testing.T) SecretsKVStore {
 	return kv
 }
 
+// Helpers to determine whether plugin startup failures are fatal to the app
+func GetNamespacedKVStore(kv kvstore.KVStore) *kvstore.NamespacedKVStore {
+	return kvstore.WithNamespace(kv, kvstore.AllOrganizations, PluginNamespace)
+}
+
+func isPluginStartupErrorFatal(ctx context.Context, kvstore *kvstore.NamespacedKVStore) (bool, error) {
```

Comment:
I would expect this function to return true when the value of the key "QuitOnPluginStartupFailureKey" is equal "true" but currently it returns true if there's an error retriving it from the secret store or if it exists, even if it's value is not "true", the value is not checked.